scripts/repo.js

```diff
@@ -0,0 +1,116 @@
+import Github from 'github';
+import request from 'request-promise';
+import { fs, tempDir, mkdirp } from 'appium-support';
+import { exec } from 'teen_process';
+import nodeFS from 'fs';
+import path from 'path';
+import unzip from 'unzip';
+import B from 'bluebird';
+import Handlebars from 'handlebars';
+import { fencedCodeTabifyDocument } from './tabs';
+
+const LANGUAGES = ['en', 'cn'];
+
+async function unzipStream (readstream, pathToUnzipped) {
+  return new B((resolve, reject) => {
+      readstream.pipe(unzip.Extract({
+        path: pathToUnzipped,
+      }));
+      readstream.on('end', resolve);
+      readstream.on('close', reject);
+  });
+}
+
+async function getRepoDocs(owner, repo, branch='master') {
+  try {
+    // Pull down a zipball of Appium repository
+    const github = new Github({
+        Promise: require('bluebird')
+    });
+    const githubBranch = await github.repos.getBranch({owner, repo, branch});
+    const ref = githubBranch.data.commit.sha;
+    const archive = await request(`https://api.github.com/repos/${owner}/${repo}/zipball/${ref}`, {
+      headers: {
+        'User-Agent': 'appium',
+      },
+      encoding: null,
+    });
+
+    // Unzip it to a temporary directory
+    const tempdir = await tempDir.openDir();
+    const pathToZip = path.resolve(tempdir, 'appium.zip');
+    await fs.writeFile(pathToZip, archive);
+    const readStream = nodeFS.createReadStream(pathToZip);
+
+    const pathToUnzipped = path.resolve(tempdir, 'appium', branch);
+    await mkdirp(pathToUnzipped);
+
+    await unzipStream(readStream, pathToUnzipped);
+
+    // Get a reference to the path to the docs
+    const pathToAppiumFiles = await fs.readdir(pathToUnzipped);
+    const pathToDocs = path.resolve(pathToUnzipped, pathToAppiumFiles[0], 'docs');
+
+    // Do some treating on the docs to get it working correctly
+    await alterDocs(pathToDocs);
+
+    return pathToDocs;
+  } catch(e) {
+    console.error('An error occurred. ', e.message);
```

Comment:
why not just avoid catching the error here and print any errors in the main invocation below?